### PR TITLE
[AAE-11496] Publish libraries to npm

### PR DESCRIPTION
**Pull request by BSekula**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by DudaRobert — APPROVED** [review]

**Comment by BSekula** [comment]
@DudaRobert thanks for the approve, I need it only to testing purpose (run travis pipelines)

File: scripts/ci/npm/publish-libs.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+do
+    echo "Publish ${PROJECT}"
+
+    cd $DIST_DIR/${PROJECT}
+
+    echo 'strict-ssl=false' >> .npmrc
+    echo 'registry=http://${NPM_REGISTRY_ADDRESS}' >> .npmrc
+    echo '//${NPM_REGISTRY_ADDRESS}/:_authToken="${NPM_REGISTRY_TOKEN}"' >> .npmrc
+
+    # npm publish
```
Comment (BSekula) [inline scripts/ci/npm/publish-libs.sh:48]: commented from obvious reason :)

File: .travis.yml [hunk 2]
```diff
@@ -35,6 +35,8 @@ stages:
     if: branch = master AND type = push
   - name: e2e
     if: type = cron || type = pull_request
+  - name: Release Libraries
+    if: (branch = develop || branch = master) AND type = push
```
Comment (mauriziovitale) [inline .travis.yml:39]: why do we need develop or master ?

Comment (mauriziovitale) [reply .travis.yml:39]: u already have 
branches:
  only:
    - master
    - develop

File: scripts/ci/npm/publish-libs.sh (hunk 1 above)
Comment (mauriziovitale) [reply scripts/ci/npm/publish-libs.sh:48]: can we have a dry run option ?

**Review by mauriziovitale — APPROVED** [review]
check the comments

File: .travis.yml (hunk 2 above)
Comment (BSekula) [reply .travis.yml:39]: Ok, so we only need to have `type = push` so we are going to publish libraries only when the develop or master is buld

File: scripts/ci/npm/publish-libs.sh (hunk 1 above)
Comment (BSekula) [reply scripts/ci/npm/publish-libs.sh]: Added variable in travis - `PUBLISH_PROJECTS`, whenever it is going to be set to false, we wont publish npm packages

File: scripts/ci/npm/publish-libs.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    case "$var" in
+      --dryrun) RUN_DRY=true
+      echo -e "Run with --dryrun, therfore script won't publish packages\n" ;;
+    esac
+done
+
+VERSION_IN_PACKAGE_JSON=`node -p "require('$ROOT_DIR/package.json')".version;`;
+
+if [ "$TRAVIS_BRANCH" = "master" ] ; then
+  NEW_LIBRATY_VARSION=VERSION_IN_PACKAGE_JSON
```
Comment (DenysVuika) [inline scripts/ci/npm/publish-libs.sh:19]: typo in the word "Library"

File: scripts/ci/npm/publish-libs.sh [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+else
+  NEW_LIBRATY_VARSION="${VERSION_IN_PACKAGE_JSON}-${TRAVIS_BUILD_NUMBER}"
```
Comment (DenysVuika) [inline scripts/ci/npm/publish-libs.sh:21]: typo in the words "Library" and "Version" (NEW_LIBRATY_VARSION)

File: scripts/ci/npm/publish-libs.sh [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+ROOT_DIR=${DIR}/../../..
+DIST_DIR=${ROOT_DIR}/dist/@alfresco
+PROJECTS_DIR=${ROOT_DIR}/projects
+
+for var in "$@"
+do
+    case "$var" in
+      --dryrun) RUN_DRY=true
+      echo -e "Run with --dryrun, therfore script won't publish packages\n" ;;
```
Comment (DenysVuika) [inline scripts/ci/npm/publish-libs.sh:12]: two semicolons in the end

Comment (DenysVuika) [inline scripts/ci/npm/publish-libs.sh:12]: should be "therefore" ?

File: scripts/ci/npm/publish-libs.sh [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+ROOT_DIR=${DIR}/../../..
+DIST_DIR=${ROOT_DIR}/dist/@alfresco
+PROJECTS_DIR=${ROOT_DIR}/projects
+
+for var in "$@"
+do
+    case "$var" in
+      --dryrun) RUN_DRY=true
```
Comment (DenysVuika) [inline scripts/ci/npm/publish-libs.sh:11]: inconsistent naming, variable called "dry run" but you use "run dry", rename to "DRY_RUN"

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: scripts/ci/npm/publish-libs.sh (hunk 4 above)
Comment (BSekula) [reply scripts/ci/npm/publish-libs.sh:21]: fixed

File: scripts/ci/npm/publish-libs.sh (hunk 5 above)
Comment (BSekula) [reply scripts/ci/npm/publish-libs.sh:12]: fixed

Comment (BSekula) [reply scripts/ci/npm/publish-libs.sh:12]: yep, fixed, lesson learned, `cspell` is not working for `sh`

File: scripts/ci/npm/publish-libs.sh (hunk 6 above)
Comment (BSekula) [reply scripts/ci/npm/publish-libs.sh:11]: Fixed

File: .travis.yml [hunk 7]
```diff
@@ -35,6 +35,8 @@ stages:
     if: branch = master AND type = push
   - name: e2e
     if: type = cron || type = pull_request
+  - name: Release Libraries
+    if: type = push
```
Comment (DenysVuika) [inline .travis.yml:39]: this should be working only for master branch

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: angular.json [hunk 8]
```diff
@@ -430,7 +430,8 @@
             "production": {
               "tsConfig": "projects/aca-shared/tsconfig.lib.prod.json"
             }
-          }
+          },
+          "defaultConfiguration": "production"
```
Comment (BSekula) [inline angular.json:434]: need to build in production, without it I got error :

```
ERROR: Trying to publish a package that has been compiled by Ivy in full compilation mode. This is not allowed.
Please delete and rebuild the package with Ivy partial compilation mode, before attempting to publish.
```

File: package.json [hunk 9]
```diff
@@ -14,6 +14,7 @@
     "prebuild": "mkdir -p ./app/.tmp && cp ./app/src/app.config.json.tpl ./app/.tmp/app.config.json",
     "build": "ng build content-ce",
     "build.release": "npm run build -- --configuration=production,release",
+    "build-libs": "ng build aca-shared",
```
Comment (BSekula) [inline package.json:17]: going to add more build in future

File: scripts/ci/npm/publish-libs.sh [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+DRY_RUN=""
+if [ "$PUBLISH_PROJECTS" = "false" ]; then
+    DRY_RUN=true
+    echo "PUBLISH_PROJECTS is set to false, run in dry mode"
+fi
+
+VERSION_IN_PACKAGE_JSON=`node -p "require('$ROOT_DIR/package.json')".version;`;
+
+if [[ $TRAVIS_BRANCH =~ ^master.*?$ ]] ; then
```
Comment (BSekula) [inline scripts/ci/npm/publish-libs.sh:16]: Copy from [ng2](https://github.com/Alfresco/alfresco-ng2-components/blob/develop/scripts/travis/release/release-npm.sh#L9-L24)

**Review by mauriziovitale — APPROVED** [review]

**Pull request merged**